node/pkg/dal/utils/utils.go

```diff
@@ -17,6 +17,23 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+func APIKeyMiddleware() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		apiKey := c.Get("X-API-Key")
+
+		validAPIKey := os.Getenv("API_KEY")
+
+		if apiKey != validAPIKey {
```

Comment:
just out of curiosity, does this mean we'll have a single api key for all users?

Reply:
yh it's just beta version, not for final milestone proof